src/osd/SnapMapper.cc

```diff
@@ -243,6 +243,7 @@ void SnapMapper::add_oid(
   MapCacher::Transaction<std::string, bufferlist> *t)
 {
   dout(20) << __func__ << " " << oid << " " << snaps << dendl;
+  assert(!snaps.empty());
```

Comment:
if this is the case, we might want to `assert(0 == "decode snaps failure")` at https://github.com/ceph/ceph/blob/master/src/osd/PG.cc#L3747 in `PG::update_snap_map()`.